src/callbackBuilder.js

```diff
@@ -30,8 +30,14 @@ function callbackBuilder(resolve, reject, options) {
         obj[variadic[i]] = arguments[i + 1];
       }
       resolve(obj);
+    } else if (variadic) {
+      const args = new Array(arguments.length - 1);
```

Comment:
How much worse would it be to do:

```
const args = Array.from(slice(arguments, 1));
```

Reply:
Disregard, I thought the problem being solved here was at the `resolve` call-site.